src/auth/connection.ts

```diff
@@ -23,10 +23,33 @@ export const ssoAccountAccessScopes = ['sso:account:access']
 export const codewhispererScopes = ['codewhisperer:completions', 'codewhisperer:analysis']
 export const defaultSsoRegion = 'us-east-1'
 
+type SsoType = 'any' | 'base' | 'builderId'
+
 export const isIamConnection = (conn?: Connection): conn is IamConnection => conn?.type === 'iam'
-export const isSsoConnection = (conn?: Connection): conn is SsoConnection => conn?.type === 'sso'
-export const isBuilderIdConnection = (conn?: Connection): conn is SsoConnection =>
-    isSsoConnection(conn) && conn.startUrl === builderIdStartUrl
+export const isSsoConnection = (conn?: Connection, type: SsoType = 'any'): conn is SsoConnection => {
+    if (conn?.type !== 'sso') {
+        return false
+    }
+    // At this point the conn is an SSO conn, but now we must determine the specific type
+    switch (type) {
+        case 'base':
+            // Any SSO type is considered an SSO connection, but in this case
+            // we only want an Identity Center connection. So we return true
+            // as long as the connection is none of the other connection types.
+
+            // IMPORTANT: This condition should grow as more SsoType's get added.
+            return !isSsoConnection(conn, 'builderId')
+        case 'builderId':
+            return conn.startUrl === builderIdStartUrl
+        case 'any':
+            return true
+    }
+}
+export const isBaseSsoConnection = (conn?: Connection): conn is SsoConnection => isSsoConnection(conn, 'base')
+export const isBuilderIdConnection = (conn?: Connection): conn is SsoConnection => isSsoConnection(conn, 'builderId')
+
+export const isValidCodeCatalystConnection = (conn: Connection): conn is SsoConnection =>
```

Comment:
Not a blocker for this PR, but `isValidCodeWhispererConnection` could also live here.

Also `getBearerToken` and `getCredentials` have no business being in the CW special-case auth module: https://github.com/aws/aws-toolkit-vscode/blob/dcf55e4d147638de82b026e31a9bef6dee70bdfa/src/codewhisperer/util/authUtil.ts#L154-L181

Probably almost nothing in the CW special-case needs to be special. Eliminating that kind of thing will reduce weird quirky behavior.

Reply:
Yeah I agree. We'll be able to simplify a lot moving stuff out of the CW specific file